### ISSUE TITLE
Strange Behaviour in Sentence Embedding Comparison

### ISSUE THREAD

**Issue by simonefrancia** [facebookresearch/LASER#72]
Hi,
as in previous issue I explained, I am using LASER embedding to compare semantic cross-lingual similarities between sentences.
It works well for the language we are facing.
I have a particular case that I cannot understand why its result is that.
This is a normal case in my pipeline

```
SOURCE SENTENCE:
LANG : en
ORIGINAL : Don't go, stay
POST-MOSES : don 't go , stay
POST-Bpe : don 't go , stay
```

```
TARGET SENTENCE:
LANG: es
ORIGINAL: No te vayas, quédate
POST-MOSES: no te vayas , quédate
POST_BPE: no te va@@ y@@ as , qu@@ é@@ date
```

```
SIMILARITY: 0.92
```
And here it's all ok: so the model is able to handle the punctuation. 
I have a case when some strange things happen.

```
SOURCE SENTENCE:
LANG : en
ORIGINAL : They say, 'Which one?'
POST-MOSES : they say , ' which one ? '
POST-Bpe : they say , ' which one ? '
```

```
TARGET SENTENCE:
LANG: es
ORIGINAL: Ellos dicen, 'Cuál?'
POST-MOSES: ellos dicen , ' cuál ? '
POST_BPE: ellos dicen , ' cuál ? '
```

```
SIMILARITY: 0.41
```

And the most strange thing is that if I remove commas both from source and target:

```
source_text : They say 'Which one?',
target_text : Ellos dicen 'Cuál?',
similarity-> 0.92
```
I could remove all the punctuation from both but I don't know if it can be a good idea, maybe it can break other cases that worked before.

**Comment by hoschwenk** [comment]
LASER is a statistical approach and we observe good performance in average for many languages.
But, we can't exclude that the results are less good in some cases. Unfortunately, it's difficult to predict when this could happen...
I wouldn't recommend to systematically remove punctuation since it was present in the training data.
Special symbols have probably a higher impact when the sentences are short, i.e. there are almost as much punctuations as words. You may try to remove them in that condition

**Comment by simonefrancia** [comment]
Ok, thanks.
Actually we decided to delete a set of "special characters" for which we have seen there are some problems and they don't produce , in our opinion, any semantic information for our cases.
```\(\)\[\]\"“…;,.:*#?!```
We decided not to erase all the punctuation because some of it can be crucial in subword matching and encoding inside BPE, for example the single quote `'`.
```don't```
```'cause```
```goin'```
Our sentence are very short ( max 8 tokens usually ).
If you have any other advice about that, we appreciate.
Thanks
